src/test/java/rs/hybridit/challenges/BookControllerTest.java

```diff
@@ -0,0 +1,124 @@
+package rs.hybridit.challenges;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import rs.hybridit.controller.BookController;
+import rs.hybridit.controller.LibraryController;
+import rs.hybridit.dto.BookDto;
+import rs.hybridit.dto.LibraryDto;
+import rs.hybridit.model.Book;
+import rs.hybridit.model.BookCopy;
+import rs.hybridit.model.Library;
+import rs.hybridit.service.BookService;
+import rs.hybridit.service.LibraryService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BookControllerTest {
+
+	@InjectMocks
+	private BookController bookController;
+
+	@Mock
+	private BookService bookService;
+
+	@Test
+	public void getBook_ExistingIdGiven_ShouldBeSuccessfull() {
+		Book b = new Book();
+		b.setId(1L);
+		when(bookService.getOne(1L)).thenReturn(b);
+		Book book = bookController.getBook(1L);
+		verify(bookService).getOne(1L);
+		assertEquals(1L, book.getId().longValue());
+	}
+
+	@Test
+	public void getBook_NonExistingIdGiven_ShouldReturnNull() {
+		Book book = bookController.getBook(1L);
+		verify(bookService).getOne(1L);
+		assertEquals(null, book);
+	}
+
+	@Test
+	public void deleteBook_ExistingIdGiven_ShouldBeSuccessfull() {
+		Book b = new Book();
+		b.setId(1L);
+		when(bookService.getOne(1L)).thenReturn(b);
+		Book book = bookController.deleteBook(1L);
+		verify(bookService).getOne(1L);
+		assertEquals(1L, book.getId().longValue());
+	}
+
+	@Test
+	public void deleteBook_NonExistingIdGiven_ShouldReturnNull() {
+		Book book = bookController.deleteBook(1L);
+		verify(bookService).getOne(1L);
+		assertEquals(null, book);
+	}
+
+	@Test
+	public void updateBook_ExistingIdGiven_ShouldBeSuccessfull() {
+		Book book = new Book();
+		book.setId(1L);
+		BookDto bookDto = new BookDto();
+		bookDto.setId(1L);
+		bookDto.setName("New book name");
+		bookDto.setAuthor("New author");
+		bookDto.setImage("C:\\Users\\Marina\\Desktop\\hybridLibrary\\HybridLibrary");
```

Comment:
No need to use such exact value in tests, but OK :D you could just used setImage("whatever");